x-pack/test/case_api_integration/security_and_spaces/tests/common/cases/resolve_case.ts

```diff
@@ -0,0 +1,221 @@
+import {
+  defaultUser,
+  postCaseReq,
+  postCaseResp,
+  postCommentUserReq,
+  getPostCaseRequest,
+} from '../../../../common/lib/mock';
+import {
+  deleteCasesByESQuery,
+  createCase,
+  resolveCase,
+  createComment,
+  removeServerGeneratedPropertiesFromCase,
+  removeServerGeneratedPropertiesFromSavedObject,
+} from '../../../../common/lib/utils';
+import {
+  secOnly,
+  obsOnly,
+  globalRead,
+  superUser,
+  secOnlyRead,
+  obsOnlyRead,
+  obsSecRead,
+  noKibanaPrivileges,
+  obsSec,
+} from '../../../../common/lib/authentication/users';
+import { getUserInfo } from '../../../../common/lib/authentication';
+
+// eslint-disable-next-line import/no-default-export
+export default ({ getService }: FtrProviderContext): void => {
+  const supertest = getService('supertest');
+  const supertestWithoutAuth = getService('supertestWithoutAuth');
+  const es = getService('es');
+
+  describe('resolve_case', () => {
+    afterEach(async () => {
+      await deleteCasesByESQuery(es);
+    });
+
+    it('should resolve a case with no comments', async () => {
+      const postedCase = await createCase(supertest, getPostCaseRequest());
+      const resolvedCase = await resolveCase({
+        supertest,
+        caseId: postedCase.id,
+        includeComments: true,
+      });
+
+      const data = removeServerGeneratedPropertiesFromCase(resolvedCase.case);
+      expect(data).to.eql(postCaseResp());
+      expect(data.comments?.length).to.eql(0);
+    });
+
+    it('should resolve a case with comments', async () => {
+      const postedCase = await createCase(supertest, postCaseReq);
+      await createComment({ supertest, caseId: postedCase.id, params: postCommentUserReq });
+      const resolvedCase = await resolveCase({
+        supertest,
+        caseId: postedCase.id,
+        includeComments: true,
+      });
+
+      const comment = removeServerGeneratedPropertiesFromSavedObject(
+        resolvedCase.case.comments![0] as AttributesTypeUser
+      );
+
+      expect(resolvedCase.case.comments?.length).to.eql(1);
+      expect(comment).to.eql({
+        type: postCommentUserReq.type,
+        comment: postCommentUserReq.comment,
+        associationType: 'case',
+        created_by: defaultUser,
+        pushed_at: null,
+        pushed_by: null,
+        updated_by: null,
+        owner: 'securitySolutionFixture',
+      });
+    });
+
+    it('should return a 400 when passing the includeSubCaseComments', async () => {
+      const { body } = await supertest
+        .get(`${CASES_URL}/case-id/resolve?includeSubCaseComments=true`)
+        .set('kbn-xsrf', 'true')
+        .send()
+        .expect(400);
+
+      expect(body.message).to.contain('disabled');
+    });
+
+    it('unhappy path - 404s when case is not there', async () => {
+      await supertest
+        .get(`${CASES_URL}/fake-id/resolve`)
+        .set('kbn-xsrf', 'true')
+        .send()
+        .expect(404);
+    });
+
+    describe('rbac', () => {
```

Comment:
Here I will add some more test cases (on another PR only for master), to reproduce the other two _resolve_ scenarios: `alilasMatch` and `conflict`